client/src/pages/proposalDetails/ProposalDetails.js

```diff
@@ -177,46 +153,35 @@ function ProposalConditionalRender(isAdmin) {
 }
 
 function ProposalDetails() {
-  const [comments, addComment] = useState(
-    [
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_LONG_COMMENT} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_MEDIUM_COMMENT} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_COMMENT1.text} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_COMMENT1.text} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_COMMENT1.text} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT1.userName} text={DUMMY_COMMENT1.text} time={DUMMY_COMMENT1.time}/>,
-      <DiscussionPost isAdmin={IS_ADMIN} userName={DUMMY_COMMENT2.userName} text={DUMMY_COMMENT2.text} time={DUMMY_COMMENT2.time}/>
-    ]
-  )
+  const [comments, setComments] = React.useState([]);
+  
+  async function fetchData() {
+    try {
+      const response = await axios.get("http://localhost:8000/get_comments", 
+                                          { params: 
+                                            { proposal_id: 1 }
+                                          });      
+      setComments(response.data);
+    } catch (error) {
+      console.error(error);
+    }
+  }
 
-  function postComment() {
-    var textBox = document.getElementById("userInputDiscussion");
-    var commentText = textBox.value;
-    if (commentText != '') {
-      textBox.value = '';
-      var currentDate = new Date();
-      var twelveHrTime = (hour) => {
-        if (hour == 0) {
-          return 12;
-        } else if (hour <= 12) {
-          return hour;
-        } else {
-          return hour - 12;
+  React.useEffect(() => {
+    fetchData();
+  }, []);
+
+
+  async function postComment() {
+    await axios({
+      method: 'post',
+      url: 'http://localhost:8000/post_comment',
+      data: {
+          comment_text: document.getElementById("userInputDiscussion").value
```

Comment:
Same comment above about using https://reactjs.org/docs/forms.html rather than `document.getElementById()`

Reply:
Same as above: We want to encourage good coding practices of adding error handling to all HTTP requests. See slack for how you can wrap this command in a try/catch block in case the POST request fails. For now, feel free to just console.log that a message occurred